### PR TITLE
Superblob support

### PR DESCRIPTION
**Pull request by lnotspotl**
This PR adds support for `superblobs` = blobs compiled for all number of shaves combined into a single binary file.

Once constructed (the only argument to the constructor is a path to a `.superblob` file), a regular `OpenVINO::Blob` struct may be generated by calling `getBlobWithNShaves(numShaves)`, whose output is a regular blob compiled for  `numShaves` shaves.

The relevant clickup task is [here](https://app.clickup.com/t/86by234am)

### CODE REVIEW

File: include/depthai/openvino/OpenVINO.hpp [hunk 1]
```diff
@@ -54,6 +54,59 @@ class OpenVINO {
...
+         */
+        Superblob(const std::string& pathToSuperblobFile);
+
+        /**
+         * @brief Generate a blob with a specific number of shaves
+         *
+         * @param nShaves: Number of shaves to generate the blob for. Must be between 1 and NUMBER_OF_PATCHES.
+         * @return dai::OpenVINO::Blob: Blob compiled for the specified number of shaves
+         */
+        dai::OpenVINO::Blob getBlobWithNShaves(int nShaves);
```
Comment (themarpe) [inline include/depthai/openvino/OpenVINO.hpp:78]:
```suggestion
        dai::OpenVINO::Blob getBlobWithNumShaves(int numShaves);
```

Nitpick, I think elsewhere `num` is used as a shorthand

File: src/openvino/OpenVINO.cpp [hunk 2]
```diff
@@ -54,6 +58,120 @@ const std::map<std::pair<std::uint32_t, std::uint32_t>, std::vector<OpenVINO::Ve
...
+    uint64_t value = *reinterpret_cast<const uint64_t*>(data);
+    return bigEndianToHost(value);
+}
+
+OpenVINO::Superblob::Superblob(const std::string& pathToSuperblobFile) {
+    data = readSuperblobFile(pathToSuperblobFile);
+    header = SuperblobHeader::fromData(data);
+}
+
+dai::OpenVINO::Blob OpenVINO::Superblob::getBlobWithNShaves(int nShaves) {
```
Comment (moratom) [inline src/openvino/OpenVINO.cpp:93]: Update name to match header (in bindings as well)

**Review by moratom — APPROVED** [review]
Other than the naming LGTM, thanks!

File: bindings/python/src/openvino/OpenVINOBindings.cpp [hunk 3]
```diff
@@ -10,6 +10,7 @@ void OpenVINOBindings::bind(pybind11::module& m, void* pCallstack){
     py::class_<OpenVINO> openvino(m, "OpenVINO", DOC(dai, OpenVINO));
     py::enum_<OpenVINO::Version> openvinoVersion(openvino, "Version", DOC(dai, OpenVINO, Version));
     py::class_<OpenVINO::Blob> openvinoBlob(openvino, "Blob", DOC(dai, OpenVINO, Blob));
+    py::class_<OpenVINO::Superblob> openvinoSuperblob(openvino, "Superblob", DOC(dai, OpenVINO, Superblob));
```
[the remark this answers is not in the document]
Comment (lnotspotl) [reply bindings/python/src/openvino/OpenVINOBindings.cpp]: I agree, it's two words combined. Changed.

File: include/depthai/openvino/OpenVINO.hpp (hunk 1 above)
Comment (themarpe) [reply include/depthai/openvino/OpenVINO.hpp]: Note, function name included for the suggested change, not just func arg name @lnotspotl

Comment (lnotspotl) [reply include/depthai/openvino/OpenVINO.hpp]: I have accidentally changed it back, my bad. Fixed in f260f3b9a8843af5135d8dfb11045c1185bbeb53. Thanks for pointing that out @themarpe

**Pull request merged**